app/src/main/java/kr/co/softcampus/sopt_assignment1/SignInActivity.kt

```diff
@@ -0,0 +1,45 @@
+package kr.co.softcampus.sopt_assignment1
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import kr.co.softcampus.sopt_assignment1.databinding.ActivitySignInBinding
+
+class SignInActivity : AppCompatActivity() {
+
+    private lateinit var binding : ActivitySignInBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignInBinding.inflate(layoutInflater)
+
+        val startForResult = registerForActivityResult(ActivityResultContracts.StartActivityForResult()){
+            result->
+            if(result.resultCode == Activity.RESULT_OK){
+                val id = result.data?.getStringExtra("id")
+                val pw = result.data?.getStringExtra("pw")
+                binding.etId.setText(id)
+                binding.etPw.setText(pw)
+            }
+        }
+
+        val intent = Intent(this, HomeActivity::class.java)
+        binding.btnLogin.setOnClickListener {
+            if(binding.etId.length()!=0 && binding.etPw.length()!=0) {
+                Toast.makeText(this, binding.etId.text.toString()+"님 환영합니다", Toast.LENGTH_SHORT).show()
```

Comment:
문자열 템플릿을 활용해서
```suggestion
                Toast.makeText(this, "${binding.etId.text}님 환영합니다", Toast.LENGTH_SHORT).show()
```
이렇게도 표현이 가능합니다 :)

Reply:
훨씬 간단하네요! 감사합니당